src/portable/nordic/nrf5x/dcd_nrf5x.c

```diff
@@ -57,6 +56,27 @@
 #include "mcu/mcu.h"
 #endif
 
+// Unfortunately there are API differences between nrfx<2.0.0 and nrfx>=2.0.0
+// Nordic actually has generated a mess here: nrfx==1.9.0 has MDK 8.40.3 while nrfx==2.0.0 has MDK 8.29.0.
+// See the below statement to catch all nrfx versions with an old API.
+#define _MDK_VERSION 10000*MDK_MAJOR_VERSION + 100*MDK_MINOR_VERSION + MDK_MICRO_VERSION
```

Comment:
unless there is a good reason (with explanation), for case where mdk conflicts, nrfx v2 should be used. If it is due to sdk tighted version, please list the sdk + mdk here.